### ISSUE TITLE
INSTextInputClient interface does not mirror NSTextInputClient protocol

### ISSUE THREAD

**Issue by abock** [xamarin/xamarin-macios#6218]
`NSTextInputClient` has a number of `@required` members, but this is not reflected in the binding... all member metadata attributed to the interface has `IsRequired = false` and as such there are [no actual managed members in the interface](https://gist.github.com/abock/0570a57bb2ea5081e5222b81d8e9039e).

As a result, [we missed implemented two of the protocol members](https://devdiv.visualstudio.com/DevDiv/_queries/edit/905845) for the new VS Mac native editor. This is now surfacing via watson crash reports.

#### [`NSTextInputClient`](https://developer.apple.com/documentation/appkit/nstextinputclient?language=objc)

```objective-c
@protocol NSTextInputClient
  @required -(void)insertText:(id _Nonnull)string replacementRange:(NSRange)replacementRange;
  @required -(void)doCommandBySelector:(SEL _Nonnull)selector;
  @required -(void)setMarkedText:(id _Nonnull)string selectedRange:(NSRange)selectedRange replacementRange:(NSRange)replacementRange;
  @required -(void)unmarkText;
  @required -(NSRange)selectedRange;
  @required -(NSRange)markedRange;
  @required -(BOOL)hasMarkedText;
  @required -(NSAttributedString * _Nullable)attributedSubstringForProposedRange:(NSRange)range actualRange:(NSRangePointer _Nullable)actualRange;
  @required -(NSArray<NSAttributedStringKey> * _Nonnull)validAttributesForMarkedText;
  @required -(NSRect)firstRectForCharacterRange:(NSRange)range actualRange:(NSRangePointer _Nullable)actualRange;
  @required -(NSUInteger)characterIndexForPoint:(NSPoint)point;

  @optional -(NSAttributedString * _Nonnull)attributedString;
  @optional -(CGFloat)fractionOfDistanceThroughGlyphForPoint:(NSPoint)point;
  @optional -(CGFloat)baselineDeltaForCharacterAtIndex:(NSUInteger)anIndex;
  @optional -(NSInteger)windowLevel;
  @optional -(BOOL)drawsVerticallyForCharacterAtIndex:(NSUInteger)charIndex __attribute__((availability(macos, introduced=10.6)));
@end
```

#### [`INSTextInputClient`](https://gist.github.com/abock/0570a57bb2ea5081e5222b81d8e9039e)
```csharp
public interface INSTextInputClient : INativeObject, IDisposable
{
}
```

**Comment by dalexsoto** [comment]
Yes, you are right, we are missing the `[Abstract]` attributes here, unfortunately this is a breaking change and can't happen until `XAMCORE_4_0` becomes a thing, that said if it happens we'll fix it.

https://github.com/xamarin/xamarin-macios/blob/5f5a584f1ee62aa8d8421d3af9d9a66e72f70b5d/src/appkit.cs#L19252-L19298

**Comment by rolfbjarne** [comment]
This was fixed in 697e5b330e3a1fc9dd05146ccae28ef591a1ee79.
